libgeopmd/src/StatsCollector.cpp

```diff
@@ -227,6 +263,46 @@ int geopm_stats_collector_report_yaml(const struct geopm_stats_collector_s *coll
     return err;
 }
 
+int geopm_stats_collector_report(const struct geopm_stats_collector_s *collector,
+                                 size_t num_requests, geopm_report_s *report)
+{
+    int err = 0;
+    try {
+        const geopm::StatsCollector *collector_cpp = reinterpret_cast<const geopm::StatsCollector *>(collector);
+        geopm::StatsCollector::report_s report_cpp = collector_cpp->report_struct();
+        if (report_cpp.metric_names.size() != num_requests) {
+            throw geopm::Exception("geopm_stats_collector_report(): Report not correctly allocated num_request provided: " +
+                                   std::to_string(num_requests) + " required: " + std::to_string(report_cpp.metric_names.size()),
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (report_cpp.host.size() >= NAME_MAX) {
+            throw geopm::Exception("geopm_stats_collector_report(): Host name too long: " + report_cpp.host,
+                            GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+        }
+        strncpy(report->host, report_cpp.host.c_str(), NAME_MAX - 1);
```

Comment:
Does the destination buffer get left unterminated if strlen == NAME_MAX-1?

`std::string::size` returns the size of a non-nul-terminated string. The longest equivalent strlen accepted by this check is NAME_MAX-1.

Then we copy:
* to report->host (big enough for strlen NAME_MAX-1 + a nul terminator)
* from report_cpp (of strlen at most NAME_MAX-1)
* of at most NAME_MAX-1 bytes, excluding any nul terminator if this limit is reached

I think writing report->host[NAME_MAX-1]='\0' after the copy would avoid that case